### ISSUE TITLE
Proposal: new panel keeping track of mobs and players present in the current room

### ISSUE THREAD

**Issue by cosmos72** [MUME/MMapper#238]
I am proposing a new feature, and if accepted I am also offering to implement it myself:

Add a new Mob/Player Panel, somewhat similar to the already existing Group Panel, that shows the list of mobs and players present in the current room.
The idea is to keep it up-to-date to the best of MMapper abilities, i.e. updating it when:
- the player moves
- some mob or player enter or leaves the room, provided that the player can see it (i.e. if MUME sends the message "Foo leaves north/south/east/west/up/down")

And also taking into account that movement of other mobs/players in the same [group](https://mume.org/help/group) as the player are omitted by MUME in many cases - for example the leader of a group receives a message like
```ACK! A pack horse didn't follow you, you lost him.```
when he/she loses a follower, but in nospam mode MUME will **not** tell if a follower successfully follows his/her leader.

Another useful feature could be:
- showing in the mob/player list what **each** one is doing, who he/she is attacking etc.

i.e. the panel is basically intended to be a snapshot of `look` or `examine`, and also updated in real-time to the best of MMapper abilities when the player or somebody else moves, or someone in the room does something significant (attacking, casting, resting, standing etc).

I am interested in feedback for the proposal above, in particular:
- does it fit MMapper features and functions, or it would be better implemented in some other program?
- does it sound useful to MUME players, or it would need some minor/major changes to be useful?

My hope is that such feature could be **useful in situations with a lot of mobs/players** - **including PK situations** - where **messages scroll very quickly and it's difficult to keep track** of who is in the room and what they are doing.

**Comment by cosmos72** [comment]
As a concrete example, the following panel
![mmapper-room-panel-mockup-v01](https://user-images.githubusercontent.com/1947270/138148364-95b8fc6e-ff87-46b0-82e3-b916fd4d9378.png)
could be shown when the command 'look' reports, after the room name:
```console
Izanami the Elf, wielding an engraved broadsword, is here, riding a horse of the Rohirrim, fighting a Dunlending trapper.
*a fearsome Troll* (X), wielding a black waraxe, is here, fighting a black snake.
A black snake is here, fighting YOU.
A Dunlending trapper is here, fighting Izanami the Elf.
A brown bear is sleeping here.
```

Also, as shown in the mockup picture, enemy players could be highlighted (in yellow or some other colors) and mobs/players fighting YOU could be highlighted too (in the example above, YOU is bold and red).

As usual, feedback is welcome :)

**Comment by cosmos72** [comment]
Now that MUME surrounds the name of every mob or player with <character>...</character>
when XML mode is enabled (which MMapper always enables and uses),
this feature can be implemented with a reasonable amount of text parsing (and some rules to match mobs long vs. short names), without using GMCP.

Do you still prefer to go the GMCP route?

**Comment by cosmos72** [comment]
I have started implementing the GUI part in a separate fork - see https://github.com/cosmos72/MMapper

**Comment by nschimme** [comment]
Take a look at: https://mume.org/help/gmcp

**Comment by Sebster7** [comment]
I've tried your beta cosmos, and it's nice, but having already seen this: https://imgur.com/pLckD9n and this https://imgur.com/8qgnEV9 (both of which were implemented by Lamia using GMCP in Mudlet), I was disappointed.  I would like there to be the room name and exits, optionally the room description, (and potentially objects when that is added to the GMCP) as well as characters - and which instance of a character the other characters are fighting, if it is possible to get this information, as it seems to be from the latter video.  This would be great for those not using Mudlet!  e.g. total newbies who are using mmapper as their client, as well as oldies that can't be bothered to migrate all their settings.  If you're on Discord, you might have seen the newbie asking if they could turn off mobs entering and leaving the room so they can read the room description.  By putting the room name, description, exits, objects and characters in a separate panel, comms in another panel, you just have the movement and fighting spam in the main window and it doesn't really matter if you miss some of it as you have the important stuff in the other panels.
